extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -228,7 +229,7 @@ export abstract class AzureAuth implements vscode.Disposable {
 		const cachedTokens = await this.getSavedTokenAdal(tenant, resource, account.key);
 
 		// Let's check to see if we can just use the cached tokens to return to the user
-		if (cachedTokens?.accessToken) {
+		if (cachedTokens?.accessToken && cachedTokens?.expiresOn) {
```

Comment:
This could be simplified to this, right? It doesn't seem to make sense to get a token but not an expiresOn value (and vice versa)

You may also want to rename it to `cachedToken` since there's only one token being returned

```suggestion
		if (cachedTokens) {
```